xdsl/irdl/constraints.py

```diff
@@ -20,6 +20,9 @@ class ConstraintContext:
     variables: dict[str, Attribute] = field(default_factory=dict)
     """The assignment of constraint variables."""
 
+    ranges: dict[str, tuple[Attribute, ...]] = field(default_factory=dict)
+    """The assignment of constraint range variables."""
+
     def copy(self):
```

Comment:
I've realised that the copy and update method now look incorrect. Am I right in thinking they should also copy/update the ranges? These functions are only used in the `AnyOf` constraint (maybe this constraint can't be used with ranges and so this is fine?